functions/configfile.pp

```diff
@@ -0,0 +1,10 @@
+# Generates configuration files for mcollective. Keys are sorted alphabetically:
+# # Configuration is managaged by puppet class main
+# key = valze
+function mcollective::configfile(Hash $confighash) >> String {
```

Comment:
I think the `configfile` vs `config_file` in the defined type is a bit confusing, lets rename this one to `hash2config`